node/resolvers/Mutations/Organizations.ts

```diff
@@ -372,15 +373,17 @@ const Organizations = {
       ctx
     )) as B2BSettingsInput
 
+    let currentOrganizationData: Organization = {} as Organization
```

Comment:
From the code here in this file I can see that  `currentOrganizationData` may end up being just `{}` instead of a real `Organization` object. `{}` is not a valid `Organization` type though, and that's probably why you had to add the casting for `as Organization`.

It's safer to be clearer here that this variable may not have a value, like this:

```suggestion
    let currentOrganizationData: Organization  | undefined
```

Then you can deal directly with this use case within `sendUpdateOrganizationMetric`, either by setting its default value to `{}` (and so the type there should be `Organization | {}` ) or by skipping the entire process in that case.

Reply:
It is OK for me, but with this approach, we need to cover more branches cases in the tests.

Reply:
@Rudge that's ok. I think we shouldn't make coverage affect the mergeability of our PRs for B2B Suite though. The tests are not very trustworthy yet, the coverage doesn't say much about their quality or real use case coverage today. 

It's better to use the types correctly, avoiding future bugs. The types are a stricter test than the tests themselves in this case. It won't allow the code to skip dealing correctly with the edge case, and that's better than covering the use case with a test that may not even do the right checks to be counted towards code coverage.